src/data/moves/move-attrs/resist-last-move-type-attr.ts

```diff
@@ -50,15 +50,15 @@ export class ResistLastMoveTypeAttr extends MoveEffectAttr {
    * Retrieve the types resisting a given type. Used by Conversion 2
    * @returns An array populated with Types, or an empty array if no resistances exist (Unknown or Stellar type)
    */
-  getTypeResistances(gameMode: GameMode, type: number): ElementalType[] {
+  getTypeResistances(gameMode: GameMode, type: ElementalType): ElementalType[] {
     const typeResistances: ElementalType[] = [];
 
-    for (let i = 0; i < Object.keys(ElementalType).length; i++) {
+    for (const e of Object.values(ElementalType)) {
```

Comment:
Please consider using a more descriptive variable name than `e`

```suggestion
    for (const elementalType of Object.values(ElementalType)) {
```